### DISCUSSION TITLE
How to build?

### DISCUSSION THREAD

**Discussion by Timwi** [gitextensions/gitextensions#9601] [Q&A] [answered]
Hello! I was looking forward to contributing a bit of dev effort to Git Extensions. Unfortunately, I couldn’t build the project and there is no information on how to do so (in any obvious spot that I could find).

Ideally, it would be nice if the git repo were in a state that prospective contributors can just clone and build. If that is not possible, then I think there should be a file (in the root of the repo) with exact instructions on how to do so after a clean clone.

I’ve read through README.md and CONTRIBUTING.md and neither provided any help. I tried both VS2019 and VS2022-Preview. Both of them complain about projects in the Externals folder missing. If I need to clone them separately into that folder, it would be nice to have a batch file that does this for me, and to mention that you need to run that batch file (or whatever other commands) in CONTRIBUTING.md or (if the instructions are extensive) a possible new BUILDING.md or COMPILING.md file.

Thanks for hearing me out. I love the program, been using it for years, and would really love to contribute.

**Comment by RussKie** [comment, marked as answer]
There's a link in our README which will take you to the [wiki](https://github.com/gitextensions/gitextensions/wiki/How-To%3A-build-instructions). I think you may have missed the [step](https://github.com/gitextensions/gitextensions/wiki/How-To%3A-build-instructions#clone) requiring to clone all the necessary submodules. Try that, and after that I think you'd get it to build.

![image](https://user-images.githubusercontent.com/4403806/135562301-ec518da4-0550-46d1-a505-735d9772ab01.png)

![image](https://user-images.githubusercontent.com/4403806/135562487-6a5697ed-b7ae-4eba-b6da-907e19d59458.png)

**Comment by Timwi** [comment]
Oh my, thank you so much. You are completely correct, I missed the big “build instructions” link in the readme.
